### PR TITLE
added event listerner for open and close modal action

### PR DESCRIPTION
**Pull request by pragyes31**

### CODE REVIEW

File: src/Component/HomePage.js [hunk 1]
```diff
@@ -1,14 +1,35 @@
...
+	constructor(props) {
+		super(props);
+		this.state = {
+			isModalOpen: false
+        };
+        this.handleModalWindow = this.handleModalWindow.bind(this);
+    }
+      handleModalWindow = e => {
+        e.preventDefault();
+        this.setState(() => ({ isModalOpen: !this.state.isModalOpen }));
```
Comment (c-ciobanu) [inline src/Component/HomePage.js]: For a better setState for this case follow [this guide ](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous).

File: src/Component/HomePage.js [hunk 2]
```diff
@@ -1,14 +1,35 @@
...
+	render() {
+		return (
+			<div>
+				<img src={logo} alt="Spli-it app logo" />
+				<button onClick={this.handleModalWindow}>Log-in</button>
+				<button onClick={this.handleModalWindow}>Sign-up</button>
+                <SignUpModal 
+                openModal={this.state.isModalOpen} 
+                closeModal={this.handleModalWindow} />
+                <SignInModal
```
Comment (c-ciobanu) [inline src/Component/HomePage.js]: The 2 modals depend on the same state so once you open one of them the other one will open too.
Is better to have separate states for each modal.

**Review by c-ciobanu — COMMENTED** [review]
In general the HomePage file has horrible indentation, the file should been formatted automatically when you committed but it did not. Re-installing your packages should resolve this since I think you didn't do it lately.

**Review by c-ciobanu — COMMENTED** [review]
Should start using [this library](https://www.npmjs.com/package/prop-types) to type check the props, this acts as documentation for the components. I think this library is already included in react-create-app so no need to install it.

File: src/Component/HomePage.js (hunk 1 above)
Comment (pragyes31) [reply src/Component/HomePage.js]: thanks for the doc. The only thing differently I did here is passing state as an argument in the callback function inside setState.

File: src/Component/HomePage.js (hunk 2 above)
Comment (pragyes31) [reply src/Component/HomePage.js]: That's a great Catch Cristi!! Added separate state for both.

**Comment by pragyes31** [comment]
@c-ciobanu I added separate state for both the modal windows, indented the Homepage code and added prop-types lib to our project.

**Comment by c-ciobanu** [comment]
@pragyes31 All good except the propTypes definition that should stay inside the signup and signin components instead of the homepage, you should define prop types only if the components uses props, in this case the homepage doesn't have props.

**Comment by pragyes31** [comment]
@c-ciobanu you're right. I added the prop type in the modal components.

**Comment by c-ciobanu** [comment]
@pragyes31 The closeModal propType should be of type func, except for this I think we are good and can merge this PR.

**Comment by pragyes31** [comment]
@c-ciobanu I added the correct prop type. Also added handleSwapModal function to HomePage.js which would enable the user to switch between sign-in and sign-up modal.
This is when let's say the user is in sigh-in modal but wishes to sign-up instead and vice versa.

**Review by c-ciobanu — APPROVED** [review]

**Pull request merged**